### PR TITLE
fix: enabled_if with `dune describe`

### LINKED ISSUE #10779: dune describe crashing with enabled_if
## Expected Behavior

Does not crash.

## Actual Behavior

Crashing with

```
Internal error, please report upstream including the contents of _build/log.
  Description:
    ("modules_and_obj_dir: failed lookup",
    { keys = []; for_ = Exe { first_exe = "test" } })
  Raised at Stdune__Code_error.raise in file
    "otherlibs/stdune/src/code_error.ml", line 10, characters 30-62
  Called from Fiber__Scheduler.exec in file "vendor/fiber/src/scheduler.ml",
    line 76, characters 8-11
  Re-raised at Stdune__Exn.raise_with_backtrace in file
    "otherlibs/stdune/src/exn.ml", line 38, characters 27-56
  Called from Fiber__Scheduler.exec in file "vendor/fiber/src/scheduler.ml",
    line 76, characters 8-11

  I must not crash. ...
  [1]
```

## Reproduction

- PR with a reproducing test: https://github.com/ocaml/dune/pull/10778

## Specifications

- Version of `dune` (output of `dune --version`): 3.16.0, `main` branch
- Version of `ocaml` (output of `ocamlc --version`): 4.14.1
- Operating system (distribution and version):  Ubuntu 22.04.4 LTS

### PR DESCRIPTION
**Pull request by moyodiallo**
Fixes #10779.

### CODE REVIEW

File: bin/describe/describe_workspace.ml [hunk 1]
```diff
@@ -562,8 +562,13 @@ module Crawl = struct
                 (Context.build_dir context)
                 (Dune_file.dir dune_file)
             in
-            let project = Dune_file.project dune_file in
-            executables sctx ~options ~project ~dir exes
+            let* expander = Super_context.expander sctx ~dir in
```
Comment (anmonteiro) [inline bin/describe/describe_workspace.ml:565]: shouldn't this rather be in the `executables` function?

Comment (moyodiallo) [reply bin/describe/describe_workspace.ml]: Done in https://github.com/ocaml/dune/pull/10881/commits/90021898996766cebaeb7324f9c176dca9f56aaf.

Comment (rgrinberg) [reply bin/describe/describe_workspace.ml]: @anmonteiro could you have another look at this PR?

**Review by anmonteiro — APPROVED** [review]

File: bin/describe/describe_workspace.ml (hunk 1 above)
Comment (anmonteiro) [reply bin/describe/describe_workspace.ml]: ah yup, looks good. just approved

**Comment by rgrinberg** [comment]
@moyodiallo needs a CHANGES entry please

**Comment by moyodiallo** [comment]
> @moyodiallo needs a CHANGES entry please

Done.

**Pull request merged**